include/node.tcc

```diff
@@ -346,6 +353,57 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_velocity_constraints() {
   }
 }
 
+//My Idea for what to input
+
+// bool assign_nodal_absorbing_constraints(double pwave_v, double swave_v);
+// !Assign absorbing constraints
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::assign_nodal_absorbing_constraint(
+    double pwave_v, double swave_v) {
+  bool status = true;
+  // Assign shear velocity constraint
+  try{
+  	unsigned dir0= 0;
```

Comment:
Be careful on where the 0 and 1. The equations talk about normal and shear, which are not necessarily the same with 0 and 1. In some cases, normal can be 0 (for instance if you have nodes on the left boundary), but normal can be 1 too (if your nodes are along the bottom of the boundary). I will explain too, but good start!